tests/test_model_upload.py

```diff
@@ -140,7 +140,7 @@ def test_model_upload_instance_with_valid_handle(self) -> None:
                     test_filepath.touch()  # Create a temporary file in the temporary directory
                     model_upload("metaresearch/new-model/pyTorch/new-variation", temp_dir, APACHE_LICENSE, "model_type")
                     self.assertEqual(len(KaggleAPIHandler.UPLOAD_BLOB_FILE_NAMES), 1)
-                    self.assertIn(TEMP_ARCHIVE_FILE, KaggleAPIHandler.UPLOAD_BLOB_FILE_NAMES)
+                    self.assertIn(TEMP_TEST_FILE, KaggleAPIHandler.UPLOAD_BLOB_FILE_NAMES)
```

Comment:
Can you add a test cases on with a folder upload and assert on the request files/directories received.

Reply:
asserting for directories will require quite a bit of changes. I will do it in a follow up pr to not complicate this one? I added a test to assert on files and an integration test that checks both files and directories